Splunk_TA_paloalto/bin/input_module_iot_security.py

```diff
@@ -0,0 +1,192 @@
+                if total < global_page_length:
+                    helper.delete_check_point("offset")
+                    helper.delete_check_point("last_run_end")
+                    helper.log_debug("End of device list. Cleared checkpoint data.")
+                    break
+            else:
+                helper.log_debug(r_status)
+                break
+        else: 
+            now = datetime.datetime.now()
+            helper.save_check_point("offset", page_offset)
+            helper.save_check_point("last_run_timestamp", datetime.datetime.strftime(now, "%Y-%m-%d %H:%M:%S"))
+            helper.log_debug("We have reached max_page. Saved offset: {0} last_run_end: {1}".format(page_offset, now))
+
+            
+    else:
+        items = 'items'
+        while total == global_page_length:
+            method = 'GET'
+            response = helper.send_http_request(url, method, parameters,
+                                                payload=None, headers=None,
+                                                cookies=None, verify=True, cert=None,
+                                                timeout=30, use_proxy=proxy_enabled)
+            r_status = response.status_code
+            if r_status == 200:
+                entries = response.json()[items]
+                results = results + entries
+                total = len(entries)
+                page_offset = page_offset + global_page_length
+                helper.log_debug("Current Offset: {0}, Total Entries: {1}".format(page_offset, total) )
+                parameters.update({'offset': page_offset})
+            else:
+                helper.log_debug(r_status)
+                break
+    run_time = time.time() - start_time
+    helper.log_debug("End of {0} results. Function took {1} to run".format(api_type, run_time))
+    return (results)
+
+def collect_events(helper, ew):
+    # Set debug level
+    log_level = helper.get_log_level()
+    helper.set_log_level(log_level)
+    # Get Proxy Settings
+    proxy_settings = helper.get_proxy()
+    proxy_enabled = bool(proxy_settings)
+    # helper.log_debug("Checking if Proxy is enabled")
+    # helper.log_debug(proxy_enabled) 
+
+    opt_customer_id = helper.get_arg('customer_id')
+    opt_access_key_id = helper.get_arg('access_key_id')
+    opt_secret_access_key = helper.get_arg('secret_access_key')
+
+    global_url = "https://{0}.iot.paloaltonetworks.com/pub/v4.0".format(
+        opt_customer_id)
+    global_url_params = {
+        'customerid': opt_customer_id,
+        'key_id': opt_access_key_id,
+        'access_key': opt_secret_access_key,
+        'pagelength': 1000,
+        'offset': 0,
+    }
+
+    last_device_pull = helper.get_check_point("last_run_timestamp")
+
+    if not last_device_pull or datetime.datetime.strptime(last_device_pull, "%Y-%m-%d %H:%M:%S") < datetime.datetime.now() - datetime.timedelta(minutes=5):
```

Comment:
Need a comment here about why we chose this criteria for performing a device fetch.